### PR TITLE
Added default file parameter options to maven_artifact

### PR DESCRIPTION
**Pull request by ademmers**
##### ISSUE TYPE
 Feature Pull Request
 
##### COMPONENT NAME

maven_artifact module

##### ANSIBLE VERSION

```
ansible 2.3.0 (devel fb32c19fea) last updated 2017/02/13 12:26:17 (GMT +200)
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### SUMMARY

This change makes use of existing common file based module parameters.
It allows for eg setting correct user, group or mode to the targte deployed artifact, without the need of an additional file task to run to correct ownership or mode (idempotence).

PRE:
```
$ ansible localhost -sKo -m maven_artifact -vvv -a 'group_id=org.jolokia artifact_id=jolokia-jvm classifier=agent dest=/tmp'
localhost | SUCCESS => {    "artifact_id": "jolokia-jvm",     "changed": true,     "classifier": "agent",     "dest": "/tmp/jolokia-jvm-latest.jar",     "extension": "jar",     "gid": 0,     "group": "root",     "group_id": "org.jolokia",     "invocation": {        "module_args": {            "artifact_id": "jolokia-jvm",             "classifier": "agent",             "dest": "/tmp",             "extension": "jar",             "group_id": "org.jolokia",             "http_agent": null,             "password": null,             "repository_url": null,             "state": "present",             "timeout": 10,             "url_password": null,             "url_username": null,             "username": null,             "validate_certs": true,             "version": "latest"        }    },     "mode": "0644",     "owner": "root",     "repository_url": "http://repo1.maven.org/maven2",     "size": 443737,     "state": "file",     "uid": 0,     "version": "latest"}
```

POST:
```
$ ansible localhost -sKo -m maven_artifact -vvv -a 'group_id=org.jolokia artifact_id=jolokia-jvm classifier=agent dest=/tmp owner=ademmers mode=0500'
localhost | SUCCESS => {    "changed": true,     "dest": "/tmp/jolokia-jvm-latest.jar",     "gid": 0,     "group": "root",     "invocation": {        "module_args": {            "artifact_id": "jolokia-jvm",             "attributes": null,             "backup": null,             "classifier": "agent",             "content": null,             "delimiter": null,             "dest": "/tmp/jolokia-jvm-latest.jar",             "directory_mode": null,             "extension": "jar",             "follow": false,             "force": null,             "group": null,             "group_id": "org.jolokia",             "http_agent": null,             "mode": "0500",             "owner": "ademmers",             "password": null,             "regexp": null,             "remote_src": null,             "repository_url": null,             "selevel": null,             "serole": null,             "setype": null,             "seuser": null,             "src": null,             "state": "present",             "timeout": 10,             "unsafe_writes": null,             "url_password": null,             "url_username": null,             "username": null,             "validate_certs": true,             "version": "latest"        }    },     "mode": "0500",     "owner": "ademmers",     "size": 443737,     "state": "file",     "uid": 1000}
```

### CODE REVIEW

File: lib/ansible/modules/packaging/language/maven_artifact.py [hunk 1]
```diff
@@ -405,13 +406,18 @@ def main():
         path = os.path.dirname(dest)
         if not os.path.exists(path):
             os.makedirs(path)
+
```
Comment (mattclay) [inline lib/ansible/modules/packaging/language/maven_artifact.py:433]: CI failure due to PEP 8 issue:

```
2017-02-13 12:09:02 ERROR: PEP 8: lib/ansible/modules/packaging/language/maven_artifact.py:409:1: W293 blank line contains whitespace (legacy)
```

**Comment by calfonso** [comment]
@ademmers just need a minor fix and rebase and we'll finish the review.

**Comment by tumbl3w33d** [comment]
Damn, this feature seems to be really popular, as this is PR number 3 for it. :trollface: 

I will try to bring #24965 through the process and get it merged.

**Comment by tumbl3w33d** [comment]
close_me

**Comment by ademmers** [comment]
out of date again ? nice, rebased once more

**Pull request closed without merging**